sputnikdao2/tests/test_quit.rs

```diff
@@ -0,0 +1,415 @@
+        dao_roles_ref(&roles),
+        vec![
+            ("all", vec![]),
+            ("council", vec![&root.account_id]),
+            ("has_nobody", vec![]),
+            ("has_2", vec![]),
+            ("has_3", vec![]),
+            ("has_23", vec![]),
+            ("has_234", vec![&user4.account_id])
+        ]
+    );
+}
+
+/// Tests a role with Ratio = 1/2 with two members,
+/// when one member votes and then the other one quits.  
+/// There should be a way for the users to "finalize"
+/// the decision on the proposal, since it would now only
+/// require that single vote.
+///
+/// https://github.com/near-daos/sputnik-dao-contract/issues/41#issuecomment-970170648
+#[test]
+fn test_quit_removes_votes1() {
+    let (root, dao) = setup_dao();
+    let user2 = root.create_user(user(2), to_yocto("1000"));
+    let user3 = root.create_user(user(3), to_yocto("1000"));
+    let user4 = root.create_user(user(4), to_yocto("1000"));
+
+    // users (2, 3) will share a role,
+    // and only user2 will vote in approval, then user3 quits.
+    // then assert that the proposals can get approved from only 1 vote.
+
+    let dao_name = {
+        let config = view!(dao.get_config()).unwrap_json::<sputnikdao2::Config>();
+        config.name
+    };
+
+    {
+        let permissions = vec!["*:*".to_string()].into_iter().collect();
+        let role_23 = new_role("has_23".to_string(), permissions);
+        policy_extend_roles(&root, &dao, vec![role_23]);
+    }
+
+    add_user_to_roles(&root, &dao, &user2, vec!["has_23"]);
+    add_user_to_roles(&root, &dao, &user3, vec!["has_23"]);
+
+    // adds two transfer proposals
+    let t1 = add_transfer_proposal(&root, &dao, &user4);
+    let t2 = add_transfer_proposal(&root, &dao, &user4);
+
+    // user2 votes in approval of both
+    vote(vec![&user2], &dao, t1);
+    vote(vec![&user2], &dao, t2);
+
+    // user3 quits role
+    let res = quit(&dao, &user3, &user3, dao_name).unwrap();
+    assert!(res);
+
+    // ok: user2 finalizes t1
+    let user4amount = user4.account().unwrap().amount;
+    call!(user2, dao.act_proposal(t1, Action::Finalize, None)).assert_success();
+    assert_eq!(
+        view!(dao.get_proposal(t1)).unwrap_json::<Proposal>().status,
+        ProposalStatus::Approved
+    );
+    // confirm user4 received the transfer
+    assert_eq!(
+        user4amount
+       // the bounty
+       + 1,
+        user4.account().unwrap().amount
+    );
+
+    // fail: user3 tries to finelize t2
+    let res = call!(user3, dao.act_proposal(t2, Action::Finalize, None));
+    should_fail_with(res, 0, "ERR_PERMISSION_DENIED");
+}
+
+/// Tests a role with Ratio = 1/2 with two members,
+/// when one member votes and then quits.  
+/// That single vote should not cause (nor allow) a state change
+/// in the proposal. That vote should be removed instead.
+///
+/// https://github.com/near-daos/sputnik-dao-contract/issues/41#issuecomment-971474598
+#[test]
+#[ignore]
+fn test_quit_removes_votes2() {
```

Comment:
This test requires the quitter's votes to be removed in order to pass, so it's currently ignored